tsdb/head_append.go

```diff
@@ -458,6 +462,33 @@ func (a *headAppender) AppendHistogram(ref storage.SeriesRef, lset labels.Labels
 	return storage.SeriesRef(s.ref), nil
 }
 
+func validateHistogram(h *histogram.Histogram) error {
+	checkSpans := func(sign string, spans []histogram.Span, buckets []int64) error {
+		var spanBuckets uint32
+		for n, span := range spans {
+			if n > 0 && span.Offset < 0 {
+				return errors.Wrap(
+					storage.ErrHistogramSpanNegativeOffset,
+					fmt.Sprintf("%s span number %d with offset %d", sign, n+1, span.Offset),
+				)
+			}
+			spanBuckets += span.Length
+		}
+		if l := uint32(len(buckets)); spanBuckets != l {
```

Comment:
With spanBuckets as int
```suggestion
		if spanBuckets != len(buckets) {
```